core-primitives/stf-executor/src/mocks.rs

```diff
@@ -28,35 +28,45 @@ use ita_stf::{
 	TrustedOperation,
 };
 use itp_sgx_externalities::SgxExternalitiesTrait;
+use itp_stf_state_handler::handle_state::HandleState;
 use itp_types::H256;
 use sp_core::Pair;
 use sp_runtime::traits::Header as HeaderTrait;
-use std::{marker::PhantomData, time::Duration, vec::Vec};
+use std::{marker::PhantomData, sync::Arc, time::Duration, vec::Vec};
 
 /// Mock for the StfExecutor.
 #[derive(Default)]
-pub struct StfExecutorMock<StateType: SgxExternalitiesTrait + Encode> {
-	_phantom: PhantomData<StateType>,
+pub struct StfExecutorMock<StateHandler> {
+	state_handler: Arc<StateHandler>,
 }
 
-impl<StateType> StateUpdateProposer for StfExecutorMock<StateType>
+impl<StateHandler> StfExecutorMock<StateHandler> {
+	pub fn new(state_handler: Arc<StateHandler>) -> Self {
+		Self { state_handler }
+	}
+}
```

Comment:
Not a big fan of this change. The whole idea of a mock is to eliminate dependencies. When we start adding functionality and dependencies to the mocks again, we defeat the purpose of a mock. A change like this usually points to a problem in encapsulation and scope of the unit test

Reply:
In order to solve the problem, I would give the `StfExecutorMock` an `Option<StateType>` that can be set. And then we can call the `prepare_state_function` on that state (if it exists). This eliminates the need for the state handler.

Reply:
Done in https://github.com/integritee-network/worker/pull/963/commits/947872f3e63b65683c9a91e2efc3132b02fa35aa. Is this how you imagined it?

Reply:
Yes, good job 👍 🚀